File: EvolutionaryGenerator_v2.1_Python/hermes/problems/kp/generator/kp_generator.py
"""Translation of ``mx.tec.hermes.problems.kp.generator.KPGenerator``."""

from hermes.problems.kp.generator.kp_individual import KPIndividual
from javacompat import Random
from metah.generator import Generator
from metah.individual import Individual


class KPGenerator(Generator):
    """Provides the methods to create knapsack problems.

    :author: Paola Azeneth Castillo Gutiérrez (Python translation)
    :original author: José Carlos Ortiz Bayliss (jcobayliss@tec.mx)
    :original version: 2.0
    """

    def __init__(self, seed: int):
        """Creates a new instance of ``KPGenerator``.

        :param seed: The seed to initialize the random number generator.
        """
        self.random = Random(seed)

    def generate(self) -> Individual:
        return KPIndividual(self.random.next_long())

File: EvolutionaryGenerator_v2.1_Java/Generator/src/mx/tec/hermes/problems/kp/generator/KPGenerator.java
package mx.tec.hermes.problems.kp.generator;

import java.util.Random;
import mx.tec.meta.Generator;
import mx.tec.meta.Individual;

/**
 * Provides the methods to create knapsack problems.
 *
 * @author José Carlos Ortiz Bayliss (jcobayliss@tec.mx)
 * @version 2.0
 */
public class KPGenerator extends Generator {

    private final Random random;

    /**
     * Creates a new instance of <code>KPGenerator</code>.
     *
     * @param seed The seed to initialize the random number generator.
     */
    public KPGenerator(long seed) {
        random = new Random(seed);
    }

    @Override
    public Individual generate() {
        return new KPIndividual(random.nextLong());
    }

}
